hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -618,13 +638,17 @@ private void updateStochasticCosts(TableName tableName, double overall, double[]
       balancer.updateStochasticCost(tableName.getNameAsString(),
         OVERALL_COST_FUNCTION_NAME, "Overall cost", overall);
 
-      // each cost function
+      // weighted imbalance
+      balancer.updateStochasticCost(tableName.getNameAsString(),
+        WIGHTED_IMBALANCE_COST_NAME, "Weighted imbalance", weightedImbalance);
+
+      // each cost function as a percent of overall cost
       for (int i = 0; i < costFunctions.size(); i++) {
         CostFunction costFunction = costFunctions.get(i);
         String costFunctionName = costFunction.getClass().getSimpleName();
         double costPercent = (overall == 0) ? 0 : (subCosts[i] / overall);
         // TODO: cost function may need a specific description
-        balancer.updateStochasticCost(tableName.getNameAsString(), costFunctionName,
+        balancer.updateStochasticCost(tableName.getNameAsString(), costFunctionName + "Percentage",
```

Comment:
unfortunately i dont think we can make this chang,e though i agree with you that it is clearer. Other users of hbase might have their own metric collection, so changing this would be a compatibility issue. Let's leave the name as-is and we can cover the clarification in our own dashboard/docs.

Reply:
That makes sense!